### PR TITLE
Support Shadowserver Vulnerable SMTP server feed, fix #1984

### LINKED ISSUE #1984: Shadowserver Parser can't differentiate between "Accessible HTTP" and "Vulnerable HTTP"
See https://github.com/certtools/intelmq/pull/1971#issue-654792490 and https://github.com/certtools/intelmq/pull/1971#discussion_r647551806

The two feeds [Accessible HTTP Report](https://www.shadowserver.org/what-we-do/network-reporting/accessible-http-report/) and [Vulnerable HTTP Report](https://www.shadowserver.org/what-we-do/network-reporting/vulnerable-http-report/) use the same file name `scan_http`. Our feed detection is solely based on the file name and detects now `Vulnerable-HTTP`. Fortunately, the columns are identical, but this still causes a wrong feed name to be added to the data, of the file is actually from the `Accessible-HTTP` feed. Shadowserver's docs say, that the `tag` column is always `http` for the Accessible HTTP feed, so the data is still differentiable maybe?

### PR DESCRIPTION
**Pull request by monoidic**
This PR adds support for the Shadowserver Vulnerable SMTP Report and fixes #1984 by differentiating between Vulnerable-HTTP and Accessible-HTTP filenames.

### CODE REVIEW

File: intelmq/bots/parsers/shadowserver/_config.py [hunk 1]
```diff
@@ -2864,6 +2890,7 @@ def scan_exchange_identifier(field):
     ('Vulnerable-ISAKMP', 'scan_isakmp', vulnerable_isakmp),
     ('Vulnerable-HTTP', 'scan_http', accessible_vulnerable_http),
     ('Vulnerable-Exchange-Server', 'scan_exchange', scan_exchange),
+    ('Vulnerable-SMTP', 'scan_smtp', vulnerable_smtp),
```
Comment (wagner-certat) [inline intelmq/bots/parsers/shadowserver/_config.py:2893]:
```suggestion
    ('Vulnerable-SMTP', 'scan_smtp_vulnerable', vulnerable_smtp),
```
In our case the filename is ` 2021-08-12-scan_smtp_vulnerable-austria-geo.csv.zip`

**Review by wagner-certat — CHANGES_REQUESTED** [review]
Could you please add an entry in `docs/user/bots.rst` in the shadowserver feed section? Thanks

otherwise ready for merge

File: intelmq/bots/parsers/shadowserver/_config.py [hunk 2]
```diff
@@ -2862,8 +2888,9 @@ def scan_exchange_identifier(field):
     ('Sinkhole-Events-HTTP-Referer IPv6', 'event6_sinkhole_http_referer', event46_sinkhole_http_referer),
     ('Spam-URL', 'spam_url', spam_url),
     ('Vulnerable-ISAKMP', 'scan_isakmp', vulnerable_isakmp),
-    ('Vulnerable-HTTP', 'scan_http', accessible_vulnerable_http),
+    ('Vulnerable-HTTP', 'scan_http_vulnerable', accessible_vulnerable_http),
```
Comment (wagner-certat) [inline intelmq/bots/parsers/shadowserver/_config.py:2891]: Good catch, thanks!

**Comment by monoidic** [comment]
The suggested improvement reminded me of #1984, so I also fixed that by making Vulnerable-HTTP match by `scan_http_vulnerable` rather than `scan_http`, and added some standalone tests for `scan_http_vulnerable`.

**Comment by wagner-certat** [comment]
> The suggested improvement reminded me of #1984,

yeah, me too - but only after you committed the other fix :)

> so I also fixed that by making Vulnerable-HTTP match by `scan_http_vulnerable` rather than `scan_http`, and added some standalone tests for Vulnerable-HTTP (previously, only Accessible-HTTP had tests).

Thanks! Didn't think that #1984 is based on such a simple mistake.

**Review by wagner-certat — APPROVED** [review]

**Comment by wagner-certat** [comment]
Thanks! Also cherry-picked for maintenance in 7a81e3709 plus changelog in b76520eac

**Pull request merged**